ga4gh/avrotools.py

```diff
@@ -23,105 +26,65 @@ class SchemaToolException(Exception):
 class SchemaTool(object):
     """
     Provides additional validation methods based on avro
-
-    Some code adapted from
-    https://github.com/apache/avro/blob/trunk/lang/py/src/avro/io.py#L101
     """
-    sinkValue = 'aRandomStringBecauseWeCanNotUseNone'
-    INT_MIN_VALUE = -(1 << 31)
-    INT_MAX_VALUE = (1 << 31) - 1
-    LONG_MIN_VALUE = -(1 << 63)
-    LONG_MAX_VALUE = (1 << 63) - 1
-
     def __init__(self, class_):
         self.class_ = class_
         self._ensureProtocolSubclass()
         self.schema = class_.schema
-        self.typeToGenerator = None
 
-    def getInvalidFields(self, jsonDict):
+    def getTypicalInstance(self):
         """
-        Takes a jsonDict instantiation of class class_.  If the jsonDict is
-        valid for that class, return an empty dictionary.  Otherwise, return a
-        multilevel dictionary of the invalid data members and only the
-        invalid data members.
+        Return a typical instance of the class
+        """
+        return self._getInstance(TypicalInstanceCreator)
 
-        This is an expensive function to run and shouldn't be on the main
-        data path.
+    def getRandomInstance(self):
         """
-        self._ensureNoExtraFields(jsonDict)
-        fields = self._getInvalidFields(self.schema, jsonDict)
-        if fields == self.sinkValue:
-            return {}
-        else:
-            return fields
+        Return a random instance of the class
+        """
+        return self._getInstance(RandomInstanceCreator)
 
-    def getTypicalInstance(self):
+    def getBogusInstance(self):
```

Comment:
Nitpick: I'd prefer 'Invalid' here to 'Bogus'. I think it would be clearer what the intention of the method is.